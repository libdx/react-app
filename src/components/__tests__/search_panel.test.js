import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import SearchPanel from '../search_panel'

describe('SearchPanel ', () => {

    let component
    let onChange

    const child = <div id="test-child-id" />

    beforeEach(() => {
        onChange = jest.fn()

        component = shallow(
            <SearchPanel onChange={onChange}>
                {child}
            </SearchPanel>
        )
    })

    it('has search input', () => {
        const searchInput = component.find('SearchInput') 
        expect(searchInput.exists()).toBe(true)
    })

    it('renders children', () => {
        const child = component.find('#test-child-id')
        expect(child.exists()).toBe(true)
    })

    it('passes correct text on change', () => {
        const searchInput = component.find('SearchInput')
    
        const newTerm = 'abcde'
        searchInput.simulate('change', newTerm)
        expect(onChange).toBeCalledWith(newTerm)
    })

    it('renders correctly', () => {
        let element = <SearchPanel>{child}</SearchPanel>
        const tree = renderer.create(element).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
