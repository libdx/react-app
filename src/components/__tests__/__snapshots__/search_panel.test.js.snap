// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel  renders correctly 1`] = `
<div
  className="search-bar"
>
  <div
    className="input-group mb-3 search-input"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value={undefined}
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
  <div
    id="test-child-id"
  />
</div>
`;
